### PR TITLE
Docstring branch 

### PR DESCRIPTION
**Pull request by apaaris**
This MR adds many docstrings to the existing methods, there is not yet 100% coverage so I will mark this as a draft while I add the missing ones.

### CODE REVIEW

File: examples/depth/train.py
Comment (bonevbs) [inline examples/depth/train.py:1]: please remove docstrings from all fucntions in `examples`, those are meant as examples and the docstrings add noise in my opinion

File: torch_harmonics/_disco_convolution.py [hunk 1]
```diff
@@ -92,11 +137,43 @@ def backward(ctx, grad_output):
 
 
 class _DiscoS2TransposeContractionCuda(torch.autograd.Function):
+    r"""
```
[the remark this answers is not in the document]
Comment (apaaris) [reply torch_harmonics/_disco_convolution.py]: Solved in `0298a51`

File: tests/test_convolution.py [hunk 2]
```diff
@@ -168,6 +161,39 @@ def _precompute_convolution_tensor_dense(
 
 @parameterized_class(("device"), _devices)
 class TestDiscreteContinuousConvolution(unittest.TestCase):
+    """
+    Test the discrete-continuous convolution module.
+
+    Parameters
+    ----------
```
Comment (azrael417) [inline tests/test_convolution.py:168]: Why do the test classes have docstrings. It is great that the individual members have those but what do the class member docstrings describe? Technically, the constructor for those classes do not get any parameters and those are just to group tests together. That being said, it would be sufficient to have a docstring like

"""

Test the discrete-continuous convolution module.
"""

And then the individual member functions have docstrings describing the individual test.

File: torch_harmonics/random_fields.py [hunk 3]
```diff
@@ -35,35 +35,6 @@
 class GaussianRandomFieldS2(torch.nn.Module):
     def __init__(self, nlat, alpha=2.0, tau=3.0, sigma=None, radius=1.0, grid="equiangular", dtype=torch.float32):
         super().__init__()
```
Comment (bonevbs) [inline torch_harmonics/random_fields.py:37]: I am curiouys why this docstring got removed?

File: torch_harmonics/filter_basis.py [hunk 4]
```diff
@@ -121,12 +116,17 @@ def __init__(
 
     @property
     def kernel_size(self):
+        """
+        Compute the kernel size for piecewise linear basis.
+        
+        Returns
+        -------
+        kernel_size: int
+            The size of the kernel
```
Comment (bonevbs) [inline torch_harmonics/filter_basis.py:125]: I would maybe specify that this is number of basis functions in the kernel

File: torch_harmonics/examples/models/sfno.py [hunk 5]
```diff
@@ -355,7 +392,15 @@ def __init__(
 
     @torch.jit.ignore
     def no_weight_decay(self):
-        return {"pos_embed", "cls_token"}
+        """
```
Comment (bonevbs) [inline torch_harmonics/examples/models/sfno.py:395]: make this a one-liner

File: torch_harmonics/convolution.py [hunk 6]
```diff
@@ -389,8 +493,28 @@ def extra_repr(self):
 
     @property
     def psi_idx(self):
+        """
```
Comment (bonevbs) [inline torch_harmonics/convolution.py:496]: one-liner

File: torch_harmonics/_disco_convolution.py [hunk 7]
```diff
@@ -42,7 +42,35 @@
 
 # some helper functions
 def _get_psi(kernel_size: int, psi_idx: torch.Tensor, psi_vals: torch.Tensor, nlat_in: int, nlon_in: int, nlat_out: int, nlon_out: int, nlat_in_local: Optional[int] = None, nlat_out_local: Optional[int] = None, semi_transposed: Optional[bool] = False):
-
+    """Creates a sparse tensor for spherical harmonic convolution operations.
```
Comment (bonevbs) [inline torch_harmonics/_disco_convolution.py:45]: one-liner

File: setup.py [hunk 8]
```diff
@@ -54,7 +54,22 @@
     warnings.warn(f"building custom extensions skipped: {e}")
 
 def get_compile_args(module_name):
```
Comment (bonevbs) [inline setup.py:56]: one-liner

File: torch_harmonics/convolution.py [hunk 9]
```diff
@@ -60,11 +60,30 @@
...
-    - "individual": for each output latitude and filter basis function the filter is numerically integrated over the sphere and normalized so that it yields 1.
-    - "mean": the norm is computed for each output latitude and then averaged over the output latitudes. Each basis function is then normalized by this mean.
+    """Normalizes convolution tensor values based on specified normalization mode.
+    
+    This function applies different normalization strategies to the convolution tensor
+    values based on the basis_norm_mode parameter. It can normalize individual basis
+    functions, compute mean normalization across all basis functions, or use support
+    weights. The function also optionally merges quadrature weights into the tensor.
+    
+    Args:
```
[the remark this answers is not in the document]
Comment (apaaris) [reply torch_harmonics/convolution.py]: Solved in `4faf8aa`

File: torch_harmonics/examples/shallow_water_equations.py [hunk 10]
```diff
@@ -115,46 +143,118 @@ def __init__(self, nlat, nlon, dt, lmax=None, mmax=None, grid="equiangular", rad
 
     def grid2spec(self, ugrid):
         """
-        spectral coefficients from spatial data
+        Convert spatial data to spectral coefficients.
```
[the remark this answers is not in the document]
Comment (apaaris) [reply torch_harmonics/examples/shallow_water_equations.py]: Corrected in `b04f119`

File: torch_harmonics/examples/models/sfno.py (hunk 5 above)
Comment (apaaris) [reply torch_harmonics/examples/models/sfno.py:395]: `62292a3`

File: torch_harmonics/convolution.py [hunk 11]
```diff
@@ -389,8 +493,28 @@ def extra_repr(self):
...
+        Get the convolution tensor index
+
+        Returns
+        -------
+        psi_idx: torch.Tensor
+            Convolution tensor index
+        """
         return torch.stack([self.psi_ker_idx, self.psi_row_idx, self.psi_col_idx], dim=0).contiguous()
 
+    def get_psi(self):
```
[the remark this answers is not in the document]
Comment (apaaris) [reply torch_harmonics/convolution.py:518]: `62292a3`

File: torch_harmonics/convolution.py (hunk 6 above)
Comment (apaaris) [reply torch_harmonics/convolution.py:508]: `62292a3`

File: tests/test_sht.py [hunk 12]
```diff
@@ -110,6 +119,26 @@ def setUp(self):
         ]
     )
     def test_sht(self, nlat, nlon, batch_size, norm, grid, tol, verbose):
+        """
```
[the remark this answers is not in the document]
Comment (apaaris) [reply tests/test_sht.py]: `62292a3`

File: torch_harmonics/examples/losses.py [hunk 13]
```diff
@@ -305,47 +387,75 @@ def __init__(self, nlat: int, nlon: int, grid: str = "equiangular"):
         self.register_buffer("k_theta_mesh", k_theta_mesh)
 
     def compute_gradients(self, x):
+        """
+        Compute gradients of the input tensor using FFT.
```
Comment (bonevbs) [inline torch_harmonics/examples/losses.py:391]: this is a bit unclear. Can you qualify that those are spatial gradients?
